### ISSUE TITLE
.drawio files are not opened by the app

### ISSUE THREAD

**Issue by DavideAG** [pawelrojek/nextcloud-drawio#140]
Hi,

After the update to drawio 1.0.0 in nextcloud 20.0.8 I can't open .drawio files. In the old version, XML files was directly recognized and automatically opened using drawio but now (after the switch to .drawio extension) they are not recognized. I changed the extension of my files from xml to drawio but nothing happens. Even creating new drawio files (in .drawio extension) it is not working prooperly.

Thank you in advance

**Comment by Entrio** [comment]
I have a similar issue, old XML files are now completely ignored but the drawio files open. One of our users is a risk assessment officer and has roughly ~200 files. Yes, we can rename them manually but i think this should have been thoroughly tested.

Guess he didn't take the risk of loosing his files into his assessment

**Comment by thomasgg23** [comment]
Can confirm this with dawio 1.0.0 and nextcloud 19.0.9.
Can't open .xml neither .drawio files.
Renaming doesn't help.

**Comment by lorettprinz** [comment]
I was able to solve it this way:
As admin go to: Settings > Additional settings > Draw.io > click save
Perhaps you must change on option like "Offline Mode" on/off. The files have to be renamed to *.drawio.

**Comment by pawelrojek** [comment]
@Entrio
There are (free) tools for batch file renaming for every OS. It shouldn't really take more than 10 minutes. You can do it server side (don't forget to "occ files:scan --all") or by installing NextCloud client on your desktop and using OS specific tools.

I had to get rid of XML file extension **override** because it was causing various problems. By default XML files are text files in Nextcloud (hence **override**), and this is an universal file format, that could be anything. It was used for legacy reasons. Draw.io didn't have any specific file extension in 2017, when this app was first released, but now it's ".drawio".

**Comment by thomasgg23** [comment]
> I was able to solve it this way:
> As admin go to: Settings > Additional settings > Draw.io > click save
> Perhaps you must change on option like "Offline Mode" on/off. The files have to be renamed to *.drawio.

worked for me

**Comment by Phaere** [comment]
rename from *.xml to *.drawio not working if filename contains non-latin characters.
diagram.xml -> diagram.drawio WORKING
диаграмма.xml -> диаграмма.drawio - DON'T WORKING
диаграмма.xml -> diagram.drawio WORKING

**Comment by Entrio** [comment]
> rename from *.xml to *.drawio not working if filename contains non-latin characters.
> diagram.xml -> diagram.drawio WORKING
> диаграмма.xml -> диаграмма.drawio - DON'T WORKING
> диаграмма.xml -> diagram.drawio WORKING

У меня тоже самое. Пришлось все переименовывать на латиницу. 

I have same issue. I ended up renaming all files manually to Latin characters. A royal pain in the ass

**Comment by osm-frasch** [comment]
> I was able to solve it this way:
> As admin go to: Settings > Additional settings > Draw.io > click save
> Perhaps you must change on option like "Offline Mode" on/off. The files have to be renamed to *.drawio.

this works for me

**Comment by Greek64** [comment]
> 
> 
> I was able to solve it this way:
> As admin go to: Settings > Additional settings > Draw.io > click save
> Perhaps you must change on option like "Offline Mode" on/off. The files have to be renamed to *.drawio.

@pawelrojek 
So, is it intended that the .drawio Files are only opened in Nextcloud if the Settings are configured for *Offline Mode*?

**Comment by depuits** [comment]
> > I was able to solve it this way:
> > As admin go to: Settings > Additional settings > Draw.io > click save
> > Perhaps you must change on option like "Offline Mode" on/off. The files have to be renamed to *.drawio.
> 
> @pawelrojek
> So, is it intended that the .drawio Files are only opened in Nextcloud if the Settings are configured for _Offline Mode_?

No, the change of the setting is only to make sure that the configuration is run. After saving it when it was changed you may set the setting back the way you like.
